tests/kernel/pending/src/pend.c

```diff
@@ -407,38 +409,44 @@ void task_monitor(void)
 	 */
 
 	TC_PRINT("Testing preemptible thread waiting on timer ...\n");
-	if (timer_end_tick != 0) {
-		TC_ERROR("Task did not pend on timer\n");
-		goto error;
-	}
+	zassert_equal(timer_end_tick, 0, "Task did not pend on timer");
 
 	/* Let the timer expire */
 	k_sleep(NUM_SECONDS(2));
 
-	if (timer_end_tick < timer_start_tick + NUM_SECONDS(1)) {
-		TC_ERROR("Task waiting on timer error\n");
-		goto error;
-	}
+	zassert_false((timer_end_tick < timer_start_tick + NUM_SECONDS(1)),
+			"Task waiting on timer error");
 
-	if (timer_data != NON_NULL_PTR) {
-		TC_ERROR("Incorrect data from timer\n");
-		goto error;
-	}
+	zassert_equal(timer_data, NON_NULL_PTR,
+				"Incorrect data from timer");
 
 	k_sem_give(&end_test_sem);
-
-	result = TC_PASS;
-
-error:
-	TC_END_RESULT(result);
-	TC_END_REPORT(result);
 }
 
-K_THREAD_DEFINE(TASK_LOW, PREEM_STACKSIZE, task_low, NULL, NULL, NULL,
-		7, 0, K_NO_WAIT);
-
-K_THREAD_DEFINE(TASK_HIGH, PREEM_STACKSIZE, task_high, NULL, NULL, NULL,
-		5, 0, K_NO_WAIT);
+void test_pending(void)
```

Comment:
why was this done this way?

seems more appropriate to move the TASK_MONITOR logic here, and leave the other two threads as K_THREAD_DEFINE()

Reply:
You are asking 
1. To remove TASK_MONITOR thread 
2. Put its logic inside test_pending
3. Remove k_sleep as well 
Right ?